BUILD.bazel

```diff
@@ -0,0 +1,34 @@
+load("@gazelle//:def.bzl", "gazelle")
+load("@rules_go//go:def.bzl", "go_binary", "go_library")
+
+# gazelle:prefix github.com/buildkite/agent/v3
+gazelle(name = "gazelle")
+
+gazelle(
+    name = "gazelle-update-repos",
+    args = [
+        "-from_file=go.mod",
+        "-to_macro=deps.bzl%go_dependencies",
+        "-prune",
+    ],
+    command = "update-repos",
+)
+
+go_binary(
+    name = "agent",
```

Comment:
We should be able to name the resulting binary `buildkite-agent` with this!

Reply:
Oh nice, that's an easy update